### PR TITLE
[MERGE] mass mailing(_sms): provide usability fixes and view improvements

### PR DESCRIPTION
**Pull request by ryv-odoo**
Smalls Improvements and bugfixes for mass_mailing(_sms).

TASK_ID : 2075733

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/mass_mailing_sms/models/mailing_mailing.py [hunk 1]
```diff
@@ -12,6 +12,14 @@
 class Mailing(models.Model):
     _inherit = 'mailing.mailing'
 
+    @api.model
+    def default_get(self, fields):
+        res = super(Mailing, self).default_get(fields)
+        print(res)
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_sms/models/mailing_mailing.py]: Leftover print statement

File: addons/mass_mailing_sms/models/mailing_mailing.py [hunk 2]
```diff
@@ -12,6 +12,14 @@
...
+        if 'mailing_type' in res and res['mailing_type'] == 'sms':
+            res['keep_archives'] = True
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_sms/models/mailing_mailing.py]: To be complete you should check that either fields is void, either keep_archives is in fields (you may ask only for some specific fields, in which case you don't want to add keep archives in it).

File: addons/mass_mailing_sms/models/mailing_mailing.py [hunk 3]
```diff
@@ -12,6 +12,14 @@
 class Mailing(models.Model):
     _inherit = 'mailing.mailing'
 
+    @api.model
+    def default_get(self, fields):
+        res = super(Mailing, self).default_get(fields)
+        print(res)
+        if 'mailing_type' in res and res['mailing_type'] == 'sms':
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_sms/models/mailing_mailing.py]: if res.get('mailing_type') == 'sms' ?

File: addons/mass_mailing_sms/wizard/sms_composer.py [hunk 4]
```diff
@@ -45,7 +45,7 @@ def _prepare_mass_sms_trace_values(self, record, sms_values):
             trace_values['ignored'] = fields.Datetime.now()
         else:
             if self.mass_sms_allow_unsubscribe:
-                sms_values['body'] = '%s %s' % ((sms_values['body'] or ''), self._get_unsubscribe_url(record.id, trace_code, sms_values['number']))
+                sms_values['body'] = '%s\nSTOP SMS : %s' % ((sms_values['body'] or ''), self._get_unsubscribe_url(record.id, trace_code, sms_values['number']))
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_sms/wizard/sms_composer.py]: This should be translatable I think.

File: addons/mass_mailing_sms/models/mailing_mailing.py (hunk 1 above)
Comment (ryv-odoo) [reply addons/mass_mailing_sms/models/mailing_mailing.py]: Oups, I am going to fix that

File: addons/mass_mailing_sms/wizard/sms_composer.py [hunk 5]
```diff
@@ -45,7 +45,7 @@ def _prepare_mass_sms_trace_values(self, record, sms_values):
             trace_values['ignored'] = fields.Datetime.now()
         else:
             if self.mass_sms_allow_unsubscribe:
-                sms_values['body'] = '%s %s' % ((sms_values['body'] or ''), self._get_unsubscribe_url(record.id, trace_code, sms_values['number']))
+                sms_values['body'] = _('%s\nSTOP SMS : %s') % ((sms_values['body'] or ''), self._get_unsubscribe_url(record.id, trace_code, sms_values['number']))
```
Comment (tde-banana-odoo) [inline addons/mass_mailing_sms/wizard/sms_composer.py]: Isn't it better
``
%s\n%s % (sms_values['body'] or '', _('STOP SMS: %s') % self._get_unsubscribe_url(record.id, trace_code, sms_values['number']))
``
?

Comment (tde-banana-odoo) [reply addons/mass_mailing_sms/wizard/sms_composer.py]: That way we correctly concatenate the body and then a translated part.

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-ff

**Pull request closed without merging**